apps/comps/components/competition-state-config.tsx

```diff
@@ -15,50 +15,40 @@ type VotingStatusConfig = {
  * "Registration is full" regardless of the time window remaining.
  */
 export function getCompetitionStateConfig(
-  competition: Competition,
+  competition: RouterOutputs["competitions"]["getById"],
   hasVoted: boolean,
 ): VotingStatusConfig {
-  const status = competition.status;
-  const compStartDate = competition.startDate
-    ? new Date(competition.startDate)
-    : null;
-  const compEndDate = competition.endDate
-    ? new Date(competition.endDate)
-    : null;
-  const votingStart = competition.votingStartDate
-    ? new Date(competition.votingStartDate)
-    : null;
-  const votingEnd = competition.votingEndDate
-    ? new Date(competition.votingEndDate)
-    : null;
-  const joinStart = competition.joinStartDate
-    ? new Date(competition.joinStartDate)
-    : null;
-  const joinEnd = competition.joinEndDate
-    ? new Date(competition.joinEndDate)
-    : null;
+  const {
+    status,
+    startDate,
+    endDate,
+    votingStartDate,
+    votingEndDate,
+    joinStartDate,
+    joinEndDate,
+  } = competition;
```

Comment:
I did go ahead and embrace the data as it comes from the competitions service (via the rpc router). I made this choice because the typescript types used on the server and client before were mismatched and definitely causing bugs. And look how much nicer this is ;)